### PR TITLE
15 write tests for howdiscoverpicker and contactsmessengerpicker

### PR DESCRIPTION
**Pull request by VitalyyP**
- rewrite the contactsMessengerPicker component and test for it;
- rewrite the howDiscoverPicker component and test for it;
- write dropdownList component and test for it.

## Release Notes

- **New Features**
	- Introduced a new `DropdownList` component for improved dropdown functionality, including keyboard navigation and ARIA accessibility.
	- Enhanced `HowDiscoverPicker` and `ContactsMessengerPicker` components with better structure and accessibility features.
	- Added a binding for the telephone number input in the `CartOrderForm` to improve user input handling.

- **Bug Fixes**
	- Improved rendering and interaction tests for dropdown components to ensure reliable user experience.

- **Style**
	- Simplified CSS for dropdown and picker components, enhancing visual clarity and reducing complexity.

- **Tests**
	- Comprehensive test suites added for `DropdownList`, `HowDiscoverPicker`, and `ContactsMessengerPicker` components to ensure functionality and reliability.

### CODE REVIEW

File: src/components/common/contactsMessengerPicker/ContactsMessengerPicker.jsx [hunk 1]
```diff
@@ -15,6 +15,8 @@ function ContactsMessengerPicker({
   language,
   remarkText,
 }) {
+  console.log('remarkText: ', remarkText);
+
```
[the remark this answers is not in the document]
Comment (VitalyyP) [reply src/components/common/contactsMessengerPicker/ContactsMessengerPicker.jsx:19]: Done

**Review by VolodymyrVorona — DISMISSED** [review]

File: src/components/common/contactsMessengerPicker/ContactsMessengerPicker.tsx [hunk 2]
```diff
@@ -70,30 +78,32 @@ function ContactsMessengerPicker({
...
             className={isOpened ? `${st.moreBtn} ${st.lessBtn}` : st.moreBtn}
             type="button"
-            aria-label="Mute volume"
+            aria-label="Messenger picker"
             onClick={() => {
               setOpened(!isOpened);
             }}
             data-cy="massager-picker"
           />
         </div>
```
[the remark this answers is not in the document]
Comment (VitalyyP) [reply src/components/common/contactsMessengerPicker/ContactsMessengerPicker.tsx]: Done

File: src/components/common/dropdownList/dropdownList.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import CustomImage from 'src/components/common/customImage';
+// import List from 'src/data/messengersList';
+
+import st from './dropDawnList.module.css';
+
+function DropdownList({ optionsList, handleSelect, pickedItem, language }) {
```
[the remark this answers is not in the document]
Comment (VitalyyP) [reply src/components/common/dropdownList/dropdownList.tsx]: Done

File: src/components/common/dropdownList/dropdownList.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import CustomImage from 'src/components/common/customImage';
+// import List from 'src/data/messengersList';
+
+import st from './dropDawnList.module.css';
```
[the remark this answers is not in the document]
Comment (VitalyyP) [reply src/components/common/dropdownList/dropdownList.tsx]: Done

**Review by yuramax — COMMENTED** [review]
A small spelling error in the file name: please replace "dropdawnList.test.tsx" with "dropdownList.test.tsx", thanks)

**Review by yuramax — DISMISSED** [review]
It works great for me.

**Review by yuramax — APPROVED** [review]
Looks good to me after the merge with main.

**Pull request merged**